src/operator/numpy/np_delete_op-inl.h

```diff
@@ -0,0 +1,313 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file np_delete_op-inl.h
+ * \brief Function definition of delete operators
+ */
+#ifndef MXNET_OPERATOR_NUMPY_NP_DELETE_OP_INL_H_
+#define MXNET_OPERATOR_NUMPY_NP_DELETE_OP_INL_H_
+
+#include <vector>
+#include <memory>
+#include <algorithm>
+#include "../../common/utils.h"
+#include "../tensor/sort_op.h"
+#include "../operator_common.h"
+#include "../mxnet_op.h"
+#include "../tensor/broadcast_reduce_op.h"
+#ifdef __CUDACC__
+#include <thrust/device_ptr.h>
+#include <thrust/device_vector.h>
+#include <thrust/functional.h>
+#include <thrust/scan.h>
+#include <thrust/sequence.h>
+#include <thrust/sort.h>
+#endif
+
+namespace mxnet {
+namespace op {
+
+struct NumpyDeleteParam : public dmlc::Parameter<NumpyDeleteParam> {
+  dmlc::optional<int> start;
+  dmlc::optional<int> stop;
+  dmlc::optional<int> step;
+  dmlc::optional<int> int_ind;
+  dmlc::optional<int> axis;
+  DMLC_DECLARE_PARAMETER(NumpyDeleteParam) {
+    DMLC_DECLARE_FIELD(start)
+    .set_default(dmlc::optional<int>())
+    .describe("If 'obj' is slice, 'start' is one of it's arguments.");
+    DMLC_DECLARE_FIELD(stop)
+    .set_default(dmlc::optional<int>())
+    .describe("If 'obj' is slice, 'stop' is one of it's arguments.");
+    DMLC_DECLARE_FIELD(step)
+    .set_default(dmlc::optional<int>())
+    .describe("If 'obj' is slice, 'step' is one of it's arguments.");
+    DMLC_DECLARE_FIELD(int_ind)
+    .set_default(dmlc::optional<int>())
+    .describe("If 'obj' is int, 'int_ind' is the index before which"
+              "'values' is inserted");
+    DMLC_DECLARE_FIELD(axis)
+    .set_default(dmlc::optional<int>())
+    .describe("Axis along which to insert `values`.");
+  }
+};
+
+namespace delete_ {
+enum DeleteOpInputs {kArr, kObj};
+enum DeleteOpOutputs {kOut};
+}  // namespace delete_
+
+struct SliceToIndices {
+  template<typename IType>
+  MSHADOW_XINLINE static void Map(int i, IType* indices, int start, int step) {
+    indices[i] = start + i * step;
+  }
+};
+
+template<typename IType>
+struct AssignNum {
+  MSHADOW_XINLINE static void Map(int i, IType* output, const IType data) {
+    output[i] = data;
+  }
+};
```

Comment:
Use the `Fill` function at: https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/init_op.h#L382 instead of this kernel.